### PR TITLE
Issue 713 RPC Properties Incorrectly Retrieved

### LINKED ISSUE #713: Fix some RPC properties being incorrectly retrieved
### Bug Report (v5.0 only)
Some getters are using incorrect methods in RPC classes. See, for example, `SDLChoice`. The required usage is:

If it’s an RPC object, use `sdl_objectForName:ofClass:`, if it’s an array of RPC objects use `sdl_objectsForName:ofClass:`, anything else (numbers, strings, enums) use `sdl_objectForName:`

##### OS & Version Information
* SDL iOS Version: release/5.0.0 branch

### PR DESCRIPTION
**Pull request by NicoleYarroch**
Fixes #713

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
Added missing test cases for fixed RPC getters.  

### Summary
* Fixed getters using incorrect methods in RPC classes. 
    * An RPC object should use `sdl_objectForName:ofClass:`
    * An array of RPC objects should use `sdl_objectsForName:ofClass:`
    * Anything else (numbers, strings, enums) should use `sdl_objectForName:`

##### Bug Fixes
* Fixed the spelling in the unit test case name `SDLGetSystemCapabilityResponseSpec`

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/SDLChangeRegistration.h [hunk 1]
```diff
@@ -3,10 +3,8 @@
 
 
 #import "SDLRPCRequest.h"
-
 #import "SDLLanguage.h"
-
-@class SDLTTSChunk;
+#import "SDLTTSChunk.h"
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLChangeRegistration.h]: This should not be imported here, it was correct previously. If necessary in the `.m` it should be imported in the `.m`

File: SmartDeviceLink/SDLSendHapticData.h [hunk 2]
```diff
@@ -8,8 +8,7 @@
 
 #import <Foundation/Foundation.h>
 #import "SDLRPCRequest.h"
-
-@class SDLHapticRect;
+#import "SDLHapticRect.h"
```
Comment (joeljfischer) [inline SmartDeviceLink/SDLSendHapticData.h]: Same as above

File: SmartDeviceLink/SDLChangeRegistration.h (hunk 1 above)
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLChangeRegistration.h]: Fixed

File: SmartDeviceLink/SDLSendHapticData.h (hunk 2 above)
Comment (NicoleYarroch) [reply SmartDeviceLink/SDLSendHapticData.h]: Fixed

**Pull request merged**